### ISSUE TITLE
Parsing error with DISTKEY ( column ) SORTKEY ( column ) in Redshift

### ISSUE THREAD

**Issue by VolkovGeoPhy** [sqlfluff/sqlfluff#5386]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

this code is unparsable
```
CREATE TABLE table_name
DISTSTYLE KEY
DISTKEY ( column )
SORTKEY ( column )
AS
(SELECT *
FROM source_table)
;

```

### Expected Behaviour

parsed sql with DISTKEY and sortkey

### Observed Behaviour

Found unparsable section

### How to reproduce

Add code and try to lint

### Dialect

redshift

### Version

2.3.5

### Configuration

doesn't matter

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by franloza** [comment]
Hi! Is this an issue? I replicated the error with the code provided

```
==== parsing violations ====
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE TABLE
                       | table_name\nDISTSTYLE KEY\nDI...'
```

However, I changed `column` (which is a [reserved](https://docs.aws.amazon.com/redshift/latest/dg/r_pg_keywords.html) word) by `column_name`, and I didn't get the error:

```sql
CREATE TABLE table_name
DISTSTYLE KEY
DISTKEY ( column_name )
SORTKEY ( column_name )
AS
(SELECT *
FROM source_table)
;
```

```
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_table_as_statement:
[L:  1, P:  1]      |            keyword:                                          'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            keyword:                                          'TABLE'
[L:  1, P: 13]      |            whitespace:                                       ' '
[L:  1, P: 14]      |            object_reference:
[L:  1, P: 14]      |                naked_identifier:                             'table_name'
[L:  1, P: 24]      |            newline:                                          '\n'
[L:  2, P:  1]      |            table_constraint:
[L:  2, P:  1]      |                keyword:                                      'DISTSTYLE'
[L:  2, P: 10]      |                whitespace:                                   ' '
[L:  2, P: 11]      |                keyword:                                      'KEY'
[L:  2, P: 14]      |                newline:                                      '\n'
[L:  3, P:  1]      |                keyword:                                      'DISTKEY'
[L:  3, P:  8]      |                whitespace:                                   ' '
[L:  3, P:  9]      |                bracketed:
[L:  3, P:  9]      |                    start_bracket:                            '('
[L:  3, P: 10]      |                    [META] indent:
[L:  3, P: 10]      |                    whitespace:                               ' '
[L:  3, P: 11]      |                    column_reference:
[L:  3, P: 11]      |                        naked_identifier:                     'column_name'
[L:  3, P: 22]      |                    whitespace:                               ' '
[L:  3, P: 23]      |                    [META] dedent:
[L:  3, P: 23]      |                    end_bracket:                              ')'
[L:  3, P: 24]      |                newline:                                      '\n'
[L:  4, P:  1]      |                keyword:                                      'SORTKEY'
[L:  4, P:  8]      |                whitespace:                                   ' '
[L:  4, P:  9]      |                bracketed:
[L:  4, P:  9]      |                    start_bracket:                            '('
[L:  4, P: 10]      |                    [META] indent:
[L:  4, P: 10]      |                    whitespace:                               ' '
[L:  4, P: 11]      |                    column_reference:
[L:  4, P: 11]      |                        naked_identifier:                     'column_name'
[L:  4, P: 22]      |                    whitespace:                               ' '
[L:  4, P: 23]      |                    [META] dedent:
[L:  4, P: 23]      |                    end_bracket:                              ')'
[L:  4, P: 24]      |            newline:                                          '\n'
[L:  5, P:  1]      |            keyword:                                          'AS'
[L:  5, P:  3]      |            newline:                                          '\n'
[L:  6, P:  1]      |            bracketed:
[L:  6, P:  1]      |                start_bracket:                                '('
[L:  6, P:  2]      |                [META] indent:
[L:  6, P:  2]      |                select_statement:
[L:  6, P:  2]      |                    select_clause:
[L:  6, P:  2]      |                        keyword:                              'SELECT'
[L:  6, P:  8]      |                        [META] indent:
[L:  6, P:  8]      |                        whitespace:                           ' '
[L:  6, P:  9]      |                        select_clause_element:
[L:  6, P:  9]      |                            wildcard_expression:
[L:  6, P:  9]      |                                wildcard_identifier:
[L:  6, P:  9]      |                                    star:                     '*'
[L:  6, P: 10]      |                        [META] dedent:
[L:  6, P: 10]      |                    newline:                                  '\n'
[L:  7, P:  1]      |                    from_clause:
[L:  7, P:  1]      |                        keyword:                              'FROM'
[L:  7, P:  5]      |                        whitespace:                           ' '
[L:  7, P:  6]      |                        from_expression:
[L:  7, P:  6]      |                            [META] indent:
[L:  7, P:  6]      |                            from_expression_element:
[L:  7, P:  6]      |                                table_expression:
[L:  7, P:  6]      |                                    table_reference:
[L:  7, P:  6]      |                                        naked_identifier:     'source_table'
[L:  7, P: 18]      |                            [META] dedent:
[L:  7, P: 18]      |                [META] dedent:
[L:  7, P: 18]      |                end_bracket:                                  ')'
[L:  7, P: 19]      |    newline:                                                  '\n'
[L:  8, P:  1]      |    statement_terminator:                                     ';'
[L:  8, P:  2]      |    newline:                                                  '\n'
[L:  9, P:  1]      |    [META] end_of_file:
```

**Comment by peterbud** [comment]
@franloza is right. If column name is NOT a reserved word, latest `sqlfluff` parses properly
